### PR TITLE
[python] add dbid to musicinfotag

### PR DESCRIPTION
**Pull request by ronie**
allow python addons to get/set the dbid of a (music) listitem.

added logging for invalid videoinfo tags (we already do this for musicinfo).

### CODE REVIEW

File: xbmc/interfaces/legacy/InfoTagMusic.h [hunk 1]
```diff
@@ -67,6 +67,24 @@ namespace XBMCAddon
...
       /// \ingroup python_InfoTagMusic
+      /// @brief \python_func{ getDbId() }
+      ///-----------------------------------------------------------------------
+      /// Get identification number of tag in database.
+      ///
+      /// @return [integer] database id
+      ///
+      ///
+      ///-----------------------------------------------------------------------
+      ///
```
Comment (phil65) [inline xbmc/interfaces/legacy/InfoTagMusic.h]: version note is missing.

Comment (phil65) [inline xbmc/interfaces/legacy/InfoTagMusic.h]: I think this returns -1 in case there is no dbid? might be worth adding a comment for that.

Comment (ronie) [inline xbmc/interfaces/legacy/InfoTagMusic.h]: wish that was all :-)
still working on it....

Comment (phil65) [inline xbmc/interfaces/legacy/InfoTagMusic.h]: ok. 👍

File: xbmc/interfaces/legacy/ListItem.cpp [hunk 2]
```diff
@@ -479,6 +479,20 @@ namespace XBMCAddon
       }
       else if (strcmpi(type, "music") == 0)
       {
+        std::string type;
+        for (InfoLabelDict::const_iterator it = infoLabels.begin(); it != infoLabels.end(); ++it)
```
Comment (phil65) [inline xbmc/interfaces/legacy/ListItem.cpp]: auto + range based for loop

File: xbmc/interfaces/legacy/ListItem.cpp [hunk 3]
```diff
@@ -479,6 +479,20 @@ namespace XBMCAddon
...
+        {
+          String key = it->first;
+          StringUtils::ToLower(key);
+          const InfoLabelValue& alt = it->second;
+          const String value(alt.which() == first ? alt.former() : emptyString);
+
+          if (key == "mediatype")
```
Comment (phil65) [inline xbmc/interfaces/legacy/ListItem.cpp]: you can combine these two checks with AND.

File: xbmc/interfaces/legacy/ListItem.cpp [hunk 4]
```diff
@@ -479,6 +479,20 @@ namespace XBMCAddon
...
+          {
+            if (CMediaTypes::IsValidMediaType(value))
+              type = value;
```
Comment (phil65) [inline xbmc/interfaces/legacy/ListItem.cpp]: You can also already set the mediatype here (setType), that way you can remove it from below.

Comment (ronie) [inline xbmc/interfaces/legacy/ListItem.cpp]: good thinking, thx!

**Comment by ronie** [comment]
updated

**Review by MartijnKaijser — APPROVED** [review]

**Comment by tamland** [comment]
I'm realizing it was a big mistake exposing dbid as int in the api as this will most likely break with the future database refactor. To be future proof it should be returned as a string.

File: xbmc/interfaces/legacy/ListItem.cpp [hunk 5]
```diff
@@ -488,7 +509,9 @@ namespace XBMCAddon
           const String value(alt.which() == first ? alt.former() : emptyString);
 
           //! @todo add the rest of the infolabels
-          if (key == "tracknumber")
+          if (key == "dbid" and !type.empty())
```
Comment (ronie) [inline xbmc/interfaces/legacy/ListItem.cpp]: lol, fixed :-)

**Comment by MartijnKaijser** [comment]
So if you want to switch it to be a string we'd better do it now before we hit RC (same for video?)

**Comment by ronie** [comment]
no idea.. json returns dbid's as an int as well, is that also a problem?

**Comment by MartijnKaijser** [comment]
Only if the new db is going to return something else as an int somewhere in the future

**Comment by ronie** [comment]
i'd say let's address all that when we get there.
'till then we should keep things unified between interfaces imo.

**Comment by tamland** [comment]
For this PR, at least remove the whole '-1' documentation as api users have no business knowing about/relying on this implementation detail.

Depending on how much is already use this, it might be worth fixing now rather than later when more addons are using it. The main issue here is that the id type is a such a low level internal thing of the database. A *slight* api misuse and you have coupled an addon to the current database.

**Comment by ronie** [comment]
ok, i'll drop the '-1' comment.

as for converting it to a string, i'm afraid i don't understand the problem.
in my simple logic dbid is an int, same like year is an int, so we return it as an int.

if we return it as a string, i can convert it back to an int in my python code, then what have we gained?

ping @phil65 since your addons are the ones that are using it currently.

**Comment by phil65** [comment]
Will adapt in case we change behaviour.

**Comment by tamland** [comment]
>as for converting it to a string, i'm afraid i don't understand the problem.
in my simple logic dbid is an int, same like year is an int, so we return it as an int.

>if we return it as a string, i can convert it back to an int in my python code, then what have we gained?

The point is: dbids are opaque identifiers, not ints. they just happens to currently be represented by ints. Using a string is just *one way* to make it very explicit that it is opaque. If you don't treat it like that and do things with it, that's your own doing. But just go ahead with this. I can make a pr to 'clarify' this api.

**Comment by ronie** [comment]
thanx for the feedback.
i'm in no rush to get this in.

will get over it again when the time comes to merge this.

**Comment by ronie** [comment]
jenkins build this please

**Pull request merged**